src/GraphQL.Tests/Bugs/Issue2932_DemoDIGraphType.cs

```diff
@@ -0,0 +1,198 @@
+    scopedCounter2: scopedCounter
+    counter3: counter
+  }
+}
+",
+                @"
+{
+  ""example"": {
+    ""id"": ""1"",
+    ""name"": ""john doe"",
+    ""children"": [""Happy"",""Dopey"",""Grumpy""],
+    ""service2Test"": 2,
+    ""counter1"": 0,
+    ""counter2"": 1,
+    ""scopedCounter1"": 0,
+    ""scopedCounter2"": 0,
+    ""counter3"": 2
+  }
+}
+");
+        }
+
+        [Fact]
+        public void DIGraphInheritsCorrectly()
+        {
+            // verifies the derived type still implements ObjectGraphType<TSourceType>
+            typeof(ObjectGraphType<SampleSource>).IsAssignableFrom(typeof(DIGraphType<SampleGraph, SampleSource>)).ShouldBeTrue();
+        }
+
+        public class TestSchema : Schema
+        {
+            public TestSchema()
+            {
+                Query = new QueryGraph();
+            }
+        }
+
+        public class QueryGraph : ObjectGraphType
+        {
+            public QueryGraph()
+            {
+                Field<NonNullGraphType<DIGraphType<SampleGraph, SampleSource>>>(
+                    "example",
+                    arguments: new QueryArguments(
+                        new QueryArgument<NonNullGraphType<IntGraphType>>() { Name = "id" },
+                        new QueryArgument<NonNullGraphType<StringGraphType>>() { Name = "name" }),
+                    resolve: context => new SampleSource { Id = context.GetArgument<int>("id"), Name = context.GetArgument<string>("name") });
+            }
+        }
+
+        public class Service1
+        {
+            private int _counter;
+
+            public Task<IEnumerable<string>> GetChildrenAsync(int sourceId, CancellationToken cancellationToken)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                return Task.FromResult<IEnumerable<string>>(sourceId switch
+                {
+                    1 => new string[] { "Happy", "Dopey", "Grumpy" },
+                    _ => throw new ArgumentOutOfRangeException()
+                });
+            }
+
+            public int Count => _counter++;
+        }
+
+        public class Service2
+        {
+            public int Value => 2;
+        }
+
+        public class SampleSource
+        {
+            public int Id;
+            public string Name = null!;
+        }
+
+        // ==== demo graph using new DIGraph<T> base class ====
+        // attributes still work
+        [Name("Sample")]
+        public class SampleGraph : DIGraph<SampleSource>
+        {
+            // standard DI injection of services
+            private readonly Service1 _service1;
+            public SampleGraph(Service1 service1)
+            {
+                _service1 = service1;
+            }
+
+            // methods work with attributes and can use source property
+            [Id]
+            public int Id() => Source.Id;
+
+            // can write as static methods so that an instance of SampleGraph is not constructed at runtime
```

Comment:
```suggestion
            // can write a static method so that an instance of SampleGraph is not constructed at runtime
```
?

Reply:
No; that would change the meaning of the sentence.